lab_3/main.py

```diff
@@ -116,7 +186,25 @@ def extract_n_grams(self, encoded_corpus: tuple) -> int:
             )
         )
         """
-        pass
+        if not isinstance(encoded_corpus, tuple):
+            return 1
+        corpus = []
+        for sentence in encoded_corpus:
+            sentences = []
+            for word in sentence:
+                words = []
+                first_symbol = 0
+
+                last_symbol = first_symbol + self.size
+                while last_symbol <= len(word):
+                    n_gram = word[first_symbol: last_symbol]
+                    first_symbol += 1
+                    last_symbol += 1
+                    words.append(tuple(n_gram))
+                sentences.append(tuple(words))
+            corpus.append(tuple(sentences))
+        self.n_grams = tuple(corpus)
+        return 0
```

Comment:
does this algorithm allows you to create n-grams? e.g. 1-grams and 4-grams?

Reply:
guess, now it does)